crates/goose/src/providers/google.rs

```diff
@@ -65,7 +66,50 @@ impl GoogleProvider {
             .send()
             .await?;
 
-        handle_response(payload, response).await
+        println!("Response from Google: {:?}", response);
+
+        // https://ai.google.dev/gemini-api/docs/troubleshooting?lang=python#error-codes
+        match response.status() {
+            StatusCode::OK => Ok(response.json().await.unwrap()),
+            StatusCode::UNAUTHORIZED | StatusCode::FORBIDDEN => {
+                Err(ProviderError::Authentication(format!("Authentication failed. Please ensure your API keys are valid and have the required permissions. \
+                    Status: {}. Response: {:?}", response.status(), response.text().await.unwrap_or_default())))
+            }
+            StatusCode::BAD_REQUEST => {
+                let status = response.status();
+                let payload: Value = response.json().await.unwrap();
+                if let Some(error) = payload.get("error") {
+                    tracing::debug!("Bad Request Error: {error:?}");
+                    print!("!!! Bad Request Error: {error:?}");
+                    let error_msg = error.get("message").unwrap().as_str().unwrap();
+                    if error_msg.to_lowercase().contains("too long") {
+                        return Err(ProviderError::ContextLengthExceeded(error_msg.to_string()));
+                    }
+                }
+                tracing::debug!(
+                    "{}", format!("Provider request failed with status: {}. Payload: {}", status, payload)
+                );
+                Err(ProviderError::RequestFailed(format!("Request failed with status: {}", status)))
+            }
+            StatusCode::INTERNAL_SERVER_ERROR => {
```

Comment:
@baxen - "probably need to check on a message to confirm this?"

Reply:
i think the GOOGLE_API_KEY you provided no longer works. i keep getting rate limited